lms/djangoapps/courseware/access_utils.py

```diff
@@ -78,5 +78,10 @@ def in_preview_mode():
     Returns whether the user is in preview mode or not.
     """
     hostname = get_current_request_hostname()
-    preview_lms_base = settings.FEATURES.get('PREVIEW_LMS_BASE', None)
-    return bool(preview_lms_base and hostname and hostname.split(':')[0] == preview_lms_base.split(':')[0])
+    port = get_current_request_port()
+    preview_lms_base = settings.FEATURES.get('PREVIEW_LMS_BASE', '')
+    preview_base = (preview_lms_base or '').split(':')
+    is_preview_hostname = bool(preview_lms_base and hostname and hostname == preview_base[0])
+    if len(preview_base) > 1:
```

Comment:
If no port is explicitly provided, there is still a port, either 80 for http, or 443 for https.  In these cases, you should check the protocol of the current request, and compare against the right value.